cake/build-settings.cake

```diff
@@ -1,23 +1,354 @@
-#load ./ci.cake
-#load ./packaging.cake
-#load ./package-checks.cake
-#load ./test-results.cake
-#load ./package-tests.cake
-#load ./package-tester.cake
-#load ./header-check.cake
-#load ./local-tasks.cake
-
-public class BuildSettings
+public static class BuildSettings
 {
-    public BuildSettings(ISetupContext context)
-    {
+	private static BuildSystem _buildSystem;
+
+	public static void Initialize(
+      // Required parameters
+        ICakeContext context,
+        string title,
+        string githubRepository,
+
+      // Optional Parameters
+        bool suppressHeaderCheck = false,
+        string[] standardHeader = null,
+        string[] exemptFiles = null,
+        
+        string solutionFile = null,
+        string[] validConfigurations = null,
+        string githubOwner = "NUnit",
+
+		bool msbuildAllowPreviewVersion = false,
+		Verbosity msbuildVerbosity = Verbosity.Minimal,
+
+        string unitTests = null, // Defaults to "**/*.tests.dll|**/*.tests.exe" (case insensitive)
+        UnitTestRunner unitTestRunner = null, // If not set, NUnitLite is used
+        string unitTestArguments = null
+        )
+	{
+        // Required arguments
         if (context == null)
-            throw new System.ArgumentNullException(nameof(context));
+            throw new ArgumentNullException(nameof(context));
+		if (title == null)
+			throw new ArgumentNullException(nameof(title));
+		if (githubRepository == null)
+			throw new ArgumentNullException(nameof(githubRepository));
+
+        Context = context;
```

Comment:
Does cake support null coalesce with a throw?

```csharp
Context = content ?? throw new ArgumentNullException(...);
```

Reply:
You make a good point. Yes it does. In the scripts - as opposed to my c# production code - I have tended to use an if because I imagine my audience (?) will understand better. That's probably wrong these days.

Fixed. I like it much better.